pyiceberg/table/__init__.py

```diff
@@ -1398,14 +1401,23 @@ class UpdateSchema:
 
     def __init__(
         self,
-        table: Table,
+        table: Optional[Table],
```

Comment:
Is this change primarily for easier testing?

Reply:
Yes, this was mostly for testing, but you can also use it without a table if you would